test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -524,6 +560,163 @@ protected static IndexReader maybeWrapReaderEs(DirectoryReader reader) throws IO
+            RangeType rangeType;
+
+            if (fieldType.typeName().equals(RangeType.DOUBLE.typeName())) {
+                start = randomDouble();
+                end = randomDoubleBetween(Math.nextUp((Double)start), Double.MAX_VALUE, false);
+                rangeType = RangeType.DOUBLE;
+            } else if (fieldType.typeName().equals(RangeType.FLOAT.typeName())) {
+                start = randomFloat();
+                end = randomFloatBetween(Math.nextUp((Float) start), Float.MAX_VALUE, false);
+                rangeType = RangeType.DOUBLE;
+            } else if (fieldType.typeName().equals(RangeType.IP.typeName())) {
+                boolean v4 = randomBoolean();
+                start = randomIp(v4);
+                end = randomIp(v4);
+                rangeType = RangeType.IP;
+            } else if (fieldType.typeName().equals(RangeType.LONG.typeName())) {
+                start = randomLong();
+                end = randomLongBetween((Long)start + 1, Long.MAX_VALUE);
+                rangeType = RangeType.LONG;
+            } else if (fieldType.typeName().equals(RangeType.INTEGER.typeName())) {
+                start = randomInt();
+                end = randomIntBetween((Integer)start, Integer.MAX_VALUE);
+                rangeType = RangeType.INTEGER;
+            } else if (fieldType.typeName().equals(RangeType.DATE.typeName())) {
+                start = randomNonNegativeLong();
+                end = randomLongBetween((Long)start + 1, Long.MAX_VALUE);
+                rangeType = RangeType.DATE;
+            } else {
+                throw new IllegalStateException("Unknown type of range [" + fieldType.typeName() + "]");
+            }
+
+            final RangeFieldMapper.Range range = new RangeFieldMapper.Range(rangeType, start, end, true, true);
+            iw.addDocument(singleton(new BinaryDocValuesField(fieldName, rangeType.encodeRanges(Collections.singleton(range)))));
+
+        } else if (fieldType.getValuesSourceType().equals(CoreValuesSourceType.BOOLEAN)) {
+            iw.addDocument(singleton(new SortedNumericDocValuesField(fieldName, randomBoolean() ? 0 : 1)));
+        } else if (fieldType.getValuesSourceType().equals(CoreValuesSourceType.IP)) {
+            boolean v4 = randomBoolean();
+            iw.addDocument(singleton(new SortedSetDocValuesField(fieldName, new BytesRef(InetAddressPoint.encode(randomIp(v4))))));
+        } else if (fieldType.getValuesSourceType().equals(CoreValuesSourceType.GEOPOINT)) {
+            iw.addDocument(singleton(new LatLonDocValuesField(fieldName, randomDouble(), randomDouble())));
+        } else {
+            throw new IllegalStateException("Unknown field type [" + fieldType.typeName() + "]");
+        }
+    }
+
+    private class MockParserContext extends Mapper.TypeParser.ParserContext {
+        MockParserContext() {
+            super(null, null, null, null, null);
+        }
+    }
+
+    private static InetAddress randomIp(boolean v4) {
```

Comment:
We've got copies of this same function in `IpRangeTests` and `IpRangeAggregatorTests`, at least.  Maybe put it up in `ESTestCase` and have all three use the same version?